geth/profiling/profiling_test.go

```diff
@@ -0,0 +1,69 @@
+package profiling
+
+import (
+	"os"
+	"testing"
+	"time"
+)
+
+func TestProfiling(t *testing.T) {
```

Comment:
Too much logic in this test. Let's split it into `TestProfiling_CPU` and `TestProfiling_Mem`. A template here is to use an arrange-act-assert template for unit tests. Obviously, in this case there're more than one of each.